File: smells/ignexcept.js
const vscode = require('vscode');

var smell = {

    detect : (token) => {

        if(token.hasOwnProperty("line")) var lineno = token.line;
        if(token.hasOwnProperty("type")) var tokenType = token.type;
        if(token.hasOwnProperty("exceptionHandler")) var handler = token.arg;

        const unwantedHandlers = ['continue','pass'];
        
        if(tokenType == "except_statement" &&  unwantedHandlers.includes(handler)) 
        {
            const warning = 'possible ignore except block at line '+ lineno;
            vscode.window.showWarningMessage(warning);
        }
    }
}

module.exports = smell;

File: py/smells/ignexcept.py
from operations.savewarnings import saveWarnings

def detect(token):
    
    if token.__contains__("line"): lineno = token["line"]
    if token.__contains__("type"): tokenType = token["type"] 
    if token.__contains__("arg"): arg = token["arg"]
    
    unwantedArgs = ['continue','pass']
        
    if tokenType == "except_statement" and arg in unwantedArgs:
            
            warning = 'ignore except block'
            saveWarnings(warning,str(lineno))
            print(warning+ ' at line '+ str(lineno))
